react/AccordionItem/AccordionItem.js

```diff
@@ -10,59 +10,65 @@ function AccordionItem({
   className,
   title,
   children,
-  open,
+  isOpen: externalIsOpen,
+  onClick,
   onOpen,
   onClose,
   ...restProps
 }) {
-  const initialHeight = open ? 'auto' : CLOSED_HEIGHT;
-  const initialVisibility = open ? 'visible' : 'hidden';
-  const initialOverflow = initialVisibility;
   const contentEl = useRef(null);
-  const [currentHeight, setCurrentHeight] = useState(initialHeight);
-  const [cssVisibility, setCssVisibility] = useState(initialVisibility);
-  const [cssOverflow, setCssOverflow] = useState(initialOverflow);
+  const [currentHeight, setCurrentHeight] = useState(CLOSED_HEIGHT);
+  const [cssVisibility, setCssVisibility] = useState('visible');
+  const [cssOverflow, setCssOverflow] = useState('visible');
+  const [cssOpacity, setCssOpacity] = useState(0);
   const [timeoutHandle, setTimeoutHandle] = useState(null);
-  const [isOpen, setIsOpen] = useState(open);
+  const [isOpen, setIsOpen] = useState(false);
+  const useInternalState = externalIsOpen === undefined;
+  const finalIsOpen = useInternalState ? isOpen : externalIsOpen;
+  const isMounting = useRef(true);
+
+  useEffect(() => {
+    if (isMounting.current === false) {
+      toggleContent({
+        el: contentEl.current,
+        isOpen: finalIsOpen,
+        onOpen,
+        onClose,
+        timeoutHandle,
+        setTimeoutHandle,
+        setCurrentHeight,
+        setCssOpacity,
+        setCssOverflow,
+        setCssVisibility
+      });
+    }
+
+    isMounting.current = false;
+  }, [finalIsOpen]);
```

Comment:
The usage of isMounting to control whether useEffect should be run feels like a code smell to me. Is there a way to do it without using the isMounting check?

Reply:
should timeoutHandle be part of the array `[finalIsOpen]` as it is only a state use inside the effect lambda?

Reply:
> The usage of isMounting to control whether useEffect should be run feels like a code smell to me. Is there a way to do it without using the isMounting check?

Currently, I don't see a tidier way of doing this. In the future we could encapsulate it in a custom hook, maybe something like :

```
useEffectOnUpdate(() => {
  // ...
});
```

Reply:
> should timeoutHandle be part of the array `[finalIsOpen]` as it is only a state use inside the effect lambda?

We only want to the lambda to run when the state of the accordion is changed, e.g. `finalIsOpen`. The timeout handler will indeed change with the accordion state, but adding it to the array is going to trigger a recursive loop.